homeassistant/components/dhcp/__init__.py

```diff
@@ -55,13 +56,25 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class DhcpServiceInfo(TypedDict):
+@dataclass
+class DhcpServiceInfo:
     """Prepared info from dhcp entries."""
 
     ip: str
     hostname: str
     macaddress: str
 
+    def __getitem__(self, name: str) -> Any:
+        """
+        Allow property access by name for compatibility reason.
+
+        Deprecated, and will be removed in version 2022.6.
+        """
+        _LOGGER.warning(
+            "__getitem__ will fail in version 2022.6. Please use <cls>.<name> instead."
```

Comment:
```suggestion
            "__getitem__ will fail in version 2022.6. Please use %s.%s instead.", self, name
```

Reply:
Might need a bit of tweaking to make it obvious what needs to change.  Probably need to play around with it a bit to get a helpful message for the dev

Reply:
I think we want to wait to add the warning until after everything in core is updated in case we aren't able to complete transitioning them all before the next release

Reply:
Should I keep the warning in, but comment it out?
Or do you prefer that it remove it altogether?

Reply:
We don't like to have commented code in the code base so 
1. Remove warning
2. Merge all integration changes
3. Put warning in

Reply:
That's what I thought: done